core/src/result.rs

```diff
@@ -31,6 +31,12 @@ pub enum Error {
     #[error("parsing failed: {0}")]
     Parser(String),
 
+    #[error("OverflowError: {0}")]
+    OverflowError(String),
+
+    #[error("failed to convert Float to Decimal: {0}")]
+    F64ToDecimalConversionError(String),
```

Comment:
```suggestion
    F64ToDecimalConversionError(Decimal),
```

How about dealing with String as Decimal?

Reply:
Shouldn't it be F64ToDecimalConversionError(F64) instead of F64ToDecimalConversionError(Decimal)?

Reply:
Oh right..! thank you 
I edited